sphinx/source/docs/user_guide/examples/extensions_example_latex.py

```diff
@@ -5,44 +5,82 @@
-    switch @model.angle_units
-      when "rad" then angle = -1 * @model.angle
-      when "deg" then angle = -1 * @model.angle * Math.PI/180.0
-
-    panel = @panel ? @plot_view.frame
-
-    xscale = @plot_view.frame.xscales[@model.x_range_name]
-    yscale = @plot_view.frame.yscales[@model.y_range_name]
-
-    sx = if @model.x_units == "data" then xscale.compute(@model.x) else panel.xview.compute(@model.x)
-    sy = if @model.y_units == "data" then yscale.compute(@model.y) else panel.yview.compute(@model.y)
-
-    sx += @model.x_offset
-    sy -= @model.y_offset
-
-    #--- End of copied section from ``Label.render`` implementation
-
-    # Must render as superpositioned div (not on canvas) so that KaTex
-    # css can properly style the text
-    @_css_text(@plot_view.canvas_view.ctx, "", sx, sy, angle)
-
-    # ``katex`` is loaded into the global window at runtime
-    # katex.renderToString returns a html ``span`` element
-    katex.render(@model.text, @el, {displayMode: true})
-
-export class LatexLabel extends Label
-  type: 'LatexLabel'
-  default_view: LatexLabelView
+import {CartesianFrame} from "models/canvas/cartesian_frame"
+
+declare const katex: any
+
+export class LatexLabelView extends LabelView {
+  model: LatexLabel
+
+  render() {
+    //--- Start of copied section from ``Label.render`` implementation
+
+    // Here because AngleSpec does units transform and label doesn't support specs
+    let angle: number
+    switch (this.model.angle_units) {
+      case "rad":
+        angle = -1 * this.model.angle
+        break
+      case "deg":
+        angle = -1 * this.model.angle * Math.PI / 180.0
+        break
+      default:
+        throw new Error("Unknown unit")
+    }
+    const panel = (this.model.panel != null) ? this.model.panel : this.plot_view.frame
+    const xscale = this.plot_view.frame.xscales[this.model.x_range_name]
+    const yscale = this.plot_view.frame.yscales[this.model.y_range_name]
+
+    let sx: number; let sy: number
+    if (this.model.x_units == "data")
+      sx = xscale.compute(this.model.x)
+    else
+      sx = panel.xview.compute(this.model.x)
+    if (this.model.x_units == "data")
+      sy = yscale.compute(this.model.y)
+    else
+      sy = panel.yview.compute(this.model.y)
+    sx += this.model.x_offset
+    sy += this.model.x_offset
+
+    //--- End of copied section from ``Label.render`` implementation
+    // Must render as superpositioned div (not on canvas) so that KaTex
+    // css can properly style the text
+    this._css_text(this.plot_view.canvas_view.ctx, "", sx, sy, angle)
+
+    // ``katex`` is loaded into the global window at runtime
+    // katex.renderToString returns a html ``span`` element
+    return katex.render(this.model.text, this.el, {displayMode: true})
+  }
+}
+
+export namespace LatexLabel {
+  export interface Attrs extends Label.Attrs {
+    panel?: CartesianFrame
+  }
+
+  export interface Props extends Label.Props {}
+}
```

Comment:
This needs an update after PR #8615. The new syntax is:
```ts
export namespace LatexLabel {
  export type Attrs = p.AttrsOf<Props>

  export type Props = Label.Props
}
```
Also, `panel?: CartesianFrame` should be needed.